packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -1314,6 +1314,29 @@ class _RenderDecoration extends RenderBox with SlottedContainerRenderObjectMixin
     return Size.zero;
   }
 
+  @override
+  void describeSemanticsConfiguration(SemanticsConfiguration config) {
+    config.merger = (Map<AbstractNode, List<SemanticsConfiguration>> childConfigs) {
+      final SemanticsMergeRule rule = SemanticsMergeRule();
+      if (prefix != null) {
+        final List<SemanticsConfiguration>? prefixMergeGroup = childConfigs
+            .remove(prefix);
+        if (prefixMergeGroup != null && prefixMergeGroup.isNotEmpty) {
```

Comment:
this seems odd, why do you have to check for both? is there any meaning behind having an empty list in `childConfigs`?

If so, this should be documented on the typedef for merger and maybe on the merger property as well.